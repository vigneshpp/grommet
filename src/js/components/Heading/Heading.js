import React, { forwardRef } from 'react';

import { StyledHeading } from './StyledHeading';

const Heading = forwardRef((
  { color, fill, level, ...rest },
  ref, // munged to avoid styled-components putting it in the DOM
) => (
  // enforce level to be a number
  <StyledHeading
    as={`h${level}`}
    colorProp={color}
    fillProp={fill}
    level={+level}
    {...rest}
    ref={ref}
  />
));

Heading.displayName = 'Heading';
Heading.defaultProps = {
  level: 1,
  responsive: true,
};

let HeadingDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  HeadingDoc = require('./doc').doc(Heading);
}
const HeadingWrapper = HeadingDoc || Heading;

export { HeadingWrapper as Heading };
