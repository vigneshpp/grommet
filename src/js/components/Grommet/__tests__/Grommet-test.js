import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';

import { hpe as hpeTheme } from 'grommet-theme-hpe';

import { Grommet } from '..';
import { Heading } from '../../Heading';
import { AnnounceContext, ResponsiveContext } from '../../../contexts';
import { grommet } from '../../../themes/grommet';

const TestAnnouncer = ({ announce }) => {
  React.useEffect(() => announce('hello', 'assertive'));
  return <div>hi</div>;
};

const customBreakpointsTheme = {
  global: {
    deviceBreakpoints: {
      phone: 'small',
      tablet: 'medium',
      computer: 'large',
    },
    breakpoints: {
      small: {
        value: 600,
      },
      medium: {
        value: 800,
      },
      large: {
        value: 1000,
      },
    },
  },
};

const SSRTester = ({ ua }) => (
  <Grommet theme={customBreakpointsTheme} userAgent={ua}>
    <ResponsiveContext.Consumer>
      {size => <Heading>{`Received size ${size} for ${ua}`}</Heading>}
    </ResponsiveContext.Consumer>
  </Grommet>
);

describe('Grommet', () => {
  afterEach(cleanup);

  test('basic', () => {
    const component = renderer.create(<Grommet />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('grommet theme', () => {
    const component = renderer.create(<Grommet theme={grommet} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('hpe theme', () => {
    const component = renderer.create(
      <Grommet theme={hpeTheme}>Grommet App</Grommet>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('themeMode', () => {
    const component = renderer.create(
      <Grommet theme={grommet} themeMode="dark" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('cssVars', () => {
    const component = renderer.create(<Grommet cssVars>Grommet App</Grommet>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('full', () => {
    const component = renderer.create(<Grommet full>Grommet App</Grommet>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('background', () => {
    const component = renderer.create(
      <Grommet full background="#0000ff">
        Grommet App
      </Grommet>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('announce', done => {
    const { container } = render(
      <Grommet>
        <AnnounceContext.Consumer>
          {announce => <TestAnnouncer announce={announce} />}
        </AnnounceContext.Consumer>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();

    // no style, no need for expectPortal
    expect(
      document.body.querySelector('#grommet-announcer[aria-live]'),
    ).toMatchSnapshot();

    setTimeout(() => {
      // should clear the aria-live container
      expect(
        document.body.querySelector('#grommet-announcer[aria-live]'),
      ).toMatchSnapshot();
      done();
    }, 600); // wait the aria-live container to clear
  });

  [
    /* eslint-disable max-len */
    'Mozilla/5.0 (iPhone; CPU iPhone OS 9_0_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13A452 Safari/601.1 PTST/396',
    'Mozilla/5.0 (iPad; CPU OS 11_2_1 like Mac OS X) AppleWebKit/604.4.7 (KHTML, like Gecko) Mobile/15C153 [FBAN/FBIOS;FBAV/156.0.0.41.97;FBBV/89172188;FBDV/iPad5,3;FBMD/iPad;FBSN/iOS;FBSV/11.2.1;FBSS/2;FBCR/;FBID/tablet;FBLC/en_GB;FBOP/5;FBRV/0]',
    'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36',
    /* eslint-enable max-len */
  ].forEach(ua => {
    test(`ssr rendering ${ua.substring(0, 25)}`, () => {
      const component = renderer.create(<SSRTester ua={ua} />);
      expect(component.toJSON()).toMatchSnapshot();
      component.unmount();
    });
  });
});
