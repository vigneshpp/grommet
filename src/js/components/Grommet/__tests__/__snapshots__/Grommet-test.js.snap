// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grommet announce 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div>
    hi
  </div>
</div>
`;

exports[`Grommet announce 2`] = `
<div
  aria-live="assertive"
  id="grommet-announcer"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
>
  hello
</div>
`;

exports[`Grommet announce 3`] = `
<div
  aria-live="assertive"
  id="grommet-announcer"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
/>
`;

exports[`Grommet background 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  background-color: #0000ff;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  className="c0"
>
  Grommet App
</div>
`;

exports[`Grommet basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;

exports[`Grommet cssVars 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  --active: rgba(221,221,221,0.5);
  --active-background: background-contrast;
  --active-text: text-strong;
  --black: #000000;
  --brand: #7D4CDB;
  --focus: #6FFFB0;
  --graph-0: accent-1;
  --graph-1: neutral-1;
  --graph-2: neutral-2;
  --graph-3: neutral-3;
  --graph-4: neutral-4;
  --placeholder: #AAAAAA;
  --selected: brand;
  --selected-background: brand;
  --selected-text: text-strong;
  --white: #FFFFFF;
  --accent-1: #6FFFB0;
  --accent-2: #FD6FFF;
  --accent-3: #81FCED;
  --accent-4: #FFCA58;
  --dark-1: #333333;
  --dark-2: #555555;
  --dark-3: #777777;
  --dark-4: #999999;
  --dark-5: #999999;
  --dark-6: #999999;
  --light-1: #F8F8F8;
  --light-2: #F2F2F2;
  --light-3: #EDEDED;
  --light-4: #DADADA;
  --light-5: #DADADA;
  --light-6: #DADADA;
  --neutral-1: #00873D;
  --neutral-2: #3D138D;
  --neutral-3: #00739D;
  --neutral-4: #A2423D;
  --status-critical: #FF4040;
  --status-error: #FF4040;
  --status-warning: #FFAA15;
  --status-ok: #00C781;
  --status-unknown: #CCCCCC;
  --status-disabled: #CCCCCC;
}

<div
  className="c0"
>
  Grommet App
</div>
`;

exports[`Grommet full 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  className="c0"
>
  Grommet App
</div>
`;

exports[`Grommet grommet theme 1`] = `
.c0 {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial,sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;

exports[`Grommet hpe theme 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 18px;
  line-height: 24px;
  background-color: #FFFFFF;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  Grommet App
</div>
`;

exports[`Grommet ssr rendering Mozilla/5.0 (X11; Linux x 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:600px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Received size large for Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36
  </h1>
</div>
`;

exports[`Grommet ssr rendering Mozilla/5.0 (iPad; CPU OS 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:600px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Received size medium for Mozilla/5.0 (iPad; CPU OS 11_2_1 like Mac OS X) AppleWebKit/604.4.7 (KHTML, like Gecko) Mobile/15C153 [FBAN/FBIOS;FBAV/156.0.0.41.97;FBBV/89172188;FBDV/iPad5,3;FBMD/iPad;FBSN/iOS;FBSV/11.2.1;FBSS/2;FBCR/;FBID/tablet;FBLC/en_GB;FBOP/5;FBRV/0]
  </h1>
</div>
`;

exports[`Grommet ssr rendering Mozilla/5.0 (iPhone; CPU  1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

@media only screen and (max-width:600px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Received size small for Mozilla/5.0 (iPhone; CPU iPhone OS 9_0_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13A452 Safari/601.1 PTST/396
  </h1>
</div>
`;

exports[`Grommet themeMode 1`] = `
.c0 {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial,sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 18px;
  line-height: 24px;
  background-color: #000000;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;
